src/test/java/hudson/plugins/deploy/tomcat/Tomcat8xAdapterTest.java

```diff
@@ -91,7 +91,9 @@ public void testVariables() throws Exception {
                 "", getVariable(usernameVariable), password);
         CredentialsProvider.lookupStores(jenkinsRule.jenkins).iterator().next().addCredentials(Domain.global(), c);
 
-        adapter = new Tomcat8xAdapter(getVariable(urlVariable), c.getId(), managerContextPath);
+        adapter =
+            new Tomcat8xAdapter(
+                getVariable(urlVariable), c.getId(), getVariable(alternativeContextVariable), managerContextPath);
```

Comment:
You need to add an environment variable if you want to use the `getVariable()` method, e.g. `envVars.put(alternativeContextVariable, alternativeContext);`.